### PR TITLE
More interactors

### PR DESCRIPTION
**Pull request by t-harps**

### CODE REVIEW

File: app/controllers/api/v1/users/approvals_controller.rb [hunk 1]
```diff
@@ -3,41 +3,41 @@ class Api::V1::Users::ApprovalsController < Api::ApiController
...
-    approval = Approval.find_by(id: params[:id], user: @user)
-    return unless approval_exists? approval
-    approvable = approval.approvable
-    type = approval.approvable_type
-    Approval.accept(@user, approvable, type)
-    render json: approval.reload
-    approvable.notify_if_subscribed('new_approval', approval_zapier_data(approval)) if type == 'Developer'
+    result = ::Users::Approvals::UpdateApproval.call(current_user: @user, params: params)
+    if result.success?
+      render json: result.approval.reload
```
Comment (dmitrytrager) [inline app/controllers/api/v1/users/approvals_controller.rb]: why 'render' is not last call here in this method?

File: app/controllers/users/approvals_controller.rb [hunk 2]
```diff
@@ -10,45 +10,36 @@ def new
   end
 
   def create
-    result = CreateApproval.call(
+    result = Users::Approvals::CreateUserApproval.call(
```
Comment (dmitrytrager) [inline app/controllers/users/approvals_controller.rb]: sometimes module is used with `::`, sometimes - without
why?

File: app/controllers/users/approvals_controller.rb [hunk 3]
```diff
@@ -10,45 +10,36 @@ def new
...
-    approvals_presenter_and_gon(approvable_type)
-    approvable.notify_if_subscribed('new_approval', approval_zapier_data(approval)) if approvable_type == 'Developer'
+  def update
+    result = Users::Approvals::UpdateApproval.call(
+      current_user: current_user,
+      params: params
+    )
+    approvals_presenter_and_gon(result.approvable_type)
+    return unless result.approvable_type == "Developer"
+    result.approvable.notify_if_subscribed("new_approval", approval_zapier_data(result.approval))
```
Comment (dmitrytrager) [inline app/controllers/users/approvals_controller.rb]: is it possible to move actions like this into interactor itself?

File: spec/controllers/api/v1/users/approvals_controller_spec.rb [hunk 4]
```diff
@@ -126,10 +126,9 @@
         expect(user.friends.include?(second_user)).to be true
       end
 
-      it 'should not be able to approve if not signed in user' do
+      it 'should not be able to approve approval belonging to another user' do
```
Comment (dmitrytrager) [inline spec/controllers/api/v1/users/approvals_controller_spec.rb]: 'it is not able'

File: spec/interactors/developers/approvals/create_approval_spec.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
+require "rails_helper"
+
+RSpec.describe Developers::Approvals::CreateApproval, type: :interactor do
+  subject(:context) { described_class.call(developer: developer, params: { user: user.email }) }
```
Comment (dmitrytrager) [inline spec/interactors/developers/approvals/create_approval_spec.rb]: can we please use another name for subject?
so we won't be confised with spec contexts

File: app/controllers/users/approvals_controller.rb (hunk 2 above)
Comment (t-harps) [inline app/controllers/users/approvals_controller.rb]: CreateApproval was not originally in any modules, i found that once I moved it into Users/Approvals you have to include the namespace in order to call it

File: app/controllers/users/approvals_controller.rb (hunk 3 above)
Comment (t-harps) [inline app/controllers/users/approvals_controller.rb]: approval_zapier_data is a shared controller method which can't be accessed from an interactor, perhaps I could move this method to some shared interactor file? Not sure where to put methods shared between interactors.

File: app/controllers/api/v1/users/approvals_controller.rb (hunk 1 above)
Comment (t-harps) [inline app/controllers/api/v1/users/approvals_controller.rb]: Was worried the return on the line after would mean the render wouldn't get called, but i guess this isn't the case, will test with it moved.

File: app/controllers/users/approvals_controller.rb (hunk 3 above)
Comment (dmitrytrager) [inline app/controllers/users/approvals_controller.rb]: If it's used mostly from controller context, let's leave it as it is.

File: app/controllers/api/v1/users/approvals_controller.rb (hunk 1 above)
Comment (dmitrytrager) [inline app/controllers/api/v1/users/approvals_controller.rb]: it wouldn't

I guess we might use something like `render '...' and return ...` in this case

**Pull request merged**